x/participation/keeper/total_allocation.go

```diff
@@ -0,0 +1,28 @@
+package keeper
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+// GetTotalAllocation returns the number of available allocations based on stakings
+func (k Keeper) GetTotalAllocation(ctx sdk.Context, addr sdk.AccAddress) (uint64, error) {
```

Comment:
IMO I would prefer to have a `string` for `addr` and return error if bech32 is invalid